services/contracts/cruize/cruize_contract.py

```diff
@@ -0,0 +1,100 @@
+import ast
+
+from decouple import config
+from components import TransactionManager
+from services import LoadContracts, DydxAdmin
+from settings_config import contract_exceptions
+from utilities.constant import (
+    WALLET_ADDRESS,
+    RINKEBY_CHAIN_ID,
+    PROBABILITY,
+    SAMPLE_SIZE,
+    MAX_WAIT_SECONDS,
+)
+from utilities.enums import ErrorCodes
+
+
+class Cruize:
+    def __init__(self):
+        self.load_contract = LoadContracts()
+        self.transaction_manager = TransactionManager()
+        contract_abi = open("services/contracts/contract_abis/Starkware.json")
+        self.contract = self.load_contract.load_contracts(
+            config("STARK_EX_CONTRACT"), contract_abi
+        )
+        self.w3 = self.load_contract.web3_provider()
+
+    # this function will deposit to cruzie contract from our  curize wallet .
+    def deposit_to_cruize(self, amount=None):
+        result = {"transaction_hash": None, "error": None}
+        amount = amount["amount"]
+
+        # need to pre calculate the price
+        # average apis call time 20s
```

Comment:
that's too high. Which api's call time are you referring to here ?

Reply:
it happens when we fetch the gas fee from the most 10 recent  blocks

Reply:
any api that will interact  with the chain and where we are using the gas strategy function to calculate the gas fee

Reply:
where is deposit_to_cruize being called from ?
the deposit api in CruizeOperations viewset right ?

Reply:
no wait, I don't this being called from CruizeOperations Viewset. So where is this being used ? Not being used yet?

Reply:
it's being called from the CruizeOperation view set ,

Reply:
Currently, we don't have a smart contract deployed on the net why we are not using this function in CruizeOperation basically it's an estimated time. when I use the get strategy  function with an APIs call it's takes time more than 20 sec that's why I write this comment  that we need to think about this